src/Replication/ReplicationStrategy.php

```diff
@@ -62,7 +62,8 @@ public function isReadOperation(CommandInterface $command)
         }
 
         if (($eval = $id === 'EVAL') || $id === 'EVALSHA') {
-            $sha1 = $eval ? sha1($command->getArgument(0)) : $command->getArgument(0);
+            $argument = $command->getArgument(0);
+            $sha1 = $eval ? sha1(strval($argument)) : $argument;
```

Comment:
What promoted this change? Is that coming from `v1.1` as well? Link?

Reply:
In PHP 8.1 passing `null` to `sha1` is not allowed any more

```
sha1(): Passing null to parameter #1 ($string) of type string is deprecated
 predis/src/Replication/ReplicationStrategy.php:65
 predis/tests/Predis/Replication/ReplicationStrategyTest.php:53
```

And this is how they solved it in `v1.1` https://github.com/predis/predis/blob/d6da14daa81b63558ffd07ab5cc6946f207f4ac4/src/Replication/ReplicationStrategy.php#L63-L68